File: SoftUni Lessons/JavaScript Development/JS Fundamental REDO 2022/06 - Data Types and Variables - More Exercise/02. Prime Number Checker.js
function primeNumberchecker (num) {
    // Check if the number is a prime number
    // If it is a prime number, return true
    // If it is not a prime number, return false
    if (num < 2) {
        console.log(false);
    } else {
        for (let i = 2; i < num; i++) {
            if (num % i === 0) {
                console.log(false);
                return;
            }
        }
        console.log(true);
    }
}

primeNumberchecker(7);  // true
primeNumberchecker(8);  // false
primeNumberchecker(81);  // false

File: SoftUni Lessons/Python Development/Python Fundamentals May 2023/06. Data Types and Variables - More Exercises/02. Prime Number Checker.py
def is_prime(n):
    if n < 2:  # 1, 0 and all negative numbers are not prime
        return False
    elif n == 2:  # 2 is prime but cannot be calculated with the formula below because of the range function
        return True
    else:
        for i in range(2, n):
            if (n % i) == 0:  # if you can precisely divide a number by another number, it is not prime
                return False
        return True


print(is_prime(int(input())))
